### ISSUE TITLE
[Request] WakeLock when viewing webcam

### ISSUE THREAD

**Issue by szh** [OctoPrint/OctoPrint#4564]
### Is your feature request related to a problem? Please describe.

When monitoring prints from my phone via a browser, the phone screen goes to sleep after the standard amount of time, instead of staying awake to do the playing video.

### Describe the solution you'd like

I would like the client code to use the [WakeLock web API](https://developer.mozilla.org/en-US/docs/Web/API/WakeLock) to signal to the device that it should stay awake. I imagine this could be added in this function:
https://github.com/OctoPrint/OctoPrint/blob/2e1ec42c5b7602be7bb57b1792598a851441c156/src/octoprint/static/js/app/viewmodels/control.js#L558

I am more than happy to implement this myself and submit a PR. The reason I'm creating an issue first is to determine if this is indeed the best place to add this or if this is even a feature that would be considered were I to create such a PR.

### Additional context

I use the [TouchUI plugin](https://github.com/BillyBlaze/OctoPrint-TouchUI). It's possible this would be better suited to that plugin.

**Comment by cp2004** [comment]
From the top of your docs link, it says it is only available in secure contexts:

![Screenshot_20220630-145210_Chrome.jpg](https://user-images.githubusercontent.com/31997505/176694878-60f1dd43-e872-4b6b-90c7-da49e61173fb.jpg)

Unfortunately that means for most OctoPrint users (& yourself included) that this would not work as the browser would not allow it.

**Comment by szh** [comment]
That's true for out of the box setup, but it's fairly easy to use a HTTPS proxy like HAProxy or use the [Ngrok Tunnel plugin](https://github.com/fieldOfView/OctoPrint-ngrok)

**Comment by jneilliii** [comment]
...or octoeverywhere plugin. I guess the question is does the api fail gracefully or not on non-https connections.

**Comment by szh** [comment]
Based on some basic testing in Chrome, it seems that `navigator.wakeLock` will simply be `undefined` on non-HTTPS pages. This would be trivial to check for.

**Comment by jneilliii** [comment]
I suspect a PR would be accepted if the feature was added/enabled as a setting, but I'm not the final say and foosel will soon be on vacation for a while.

**Comment by szh** [comment]
I created a draft PR. Now I need to get a dev environment working so I can test it.
